File: rust/typical90/src/bin/typical90_cd.rs
fn main(){
	proconio::input!{
		(l,r):(i128,i128)
	}
	let nl =l.to_string().len() as i128;
	let nr =r.to_string().len() as i128;
	const MOD:i128 = 1000_000_007;
	let mut ans = 0;
	let mut left = l;
	for i in nl..=nr {
		let d = 10i128.pow(i as u32)-1;
		let right = r.min(d);
		let n = right-left+1;
		let mut t = left+right;
		t *= n; t /= 2; t %= MOD;
		t *= i; t %= MOD;
		ans += t; ans %= MOD;
		left = right+1;
	}
    println!("{}",ans);
}

File: python/typical90/typical90_cd.py
l,r=input().split()
nl=len(l)
nr=len(r)
left=int(l)
r=int(r)
MOD=10**9+7
ans=0
for i in range(nl,nr+1):
	d=int('9'*i)
	right=min(r,d)
	n=right-left+1
	t=(left+right)*n//2%MOD
	t*=i
	t%=MOD
	ans+=t
	ans%=MOD
	left=right+1
print(ans)
